docs/source/developers/python.rst

```diff
@@ -427,12 +427,9 @@ libraries), one can set ``--bundle-arrow-cpp``:
 .. code-block::
 
    $ pip install wheel  # if not installed
-   $ python setup.py build_ext --build-type=$ARROW_BUILD_TYPE \
-            --bundle-arrow-cpp bdist_wheel
-
-.. note::
-   To install an editable PyArrow build run ``pip install -e . --no-build-isolation``
-   in the ``arrow/python`` directory.
+   $ python -m pip wheel . \
```

Comment:
```suggestion
   $ python -m build . \
```

Using https://build.pypa.io/en/stable/ might be the recommended way nowadays to build a wheel?

When using that, it might need `--config-setting` instead of `--global-option` (but haven't verified this works the same)

Reply:
> When using that, it might need `--config-setting` instead of `--global-option` (but haven't verified this works the same)

I believe the `--global-option` is a requirement as long as setuptools is the backend. It's extremely murky, but here's the best conversation I could find on that topic:

https://github.com/pypa/pip/issues/11859#issuecomment-1778620671

Reply:
Yes, we need some way to pass down the option, but what I meant is that when using `python -m build` instead of `python -m pip wheel`, the name of the keyword to pass options might be different.

Because in the docs of `build`, I only see `--config-setting` and not `--global-option`